### PR TITLE
Added: Tests to the simple functions we have

### PR DESCRIPTION
**Pull request by xescugc**
Those are not all the tests that are missing but are basically all the tools around terracognita itself.

### CODE REVIEW

File: writer/state_test.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+}
+
+func TestTFStateWriter_Sync(t *testing.T) {
+	t.Run("Success", func(t *testing.T) {
+		var (
+			b     = &bytes.Buffer{}
+			sw    = writer.NewTFStateWriter(b)
+			value = &terraform.ResourceState{
+				Type: "my_type",
+			}
```
Comment (xlr-8) [inline writer/state_test.go]: Shouldn't we try writting something more complex?

File: tag/tag_test.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+			Name:     "WithTags",
+			Key:      "tags",
+			SRD:      createSRD(t, "tags", tagKey, "res"),
+			Fallback: "fallback",
+			Result:   "res",
+		},
+		{
+			Name:     "WithTagsButInvalidName",
+			Key:      "tags",
+			SRD:      createSRD(t, "tags", tagKey, "res.res"),
```
[the remark this answers is not in the document]
Comment (xescugc) [reply tag/tag_test.go]: This `Name` I'm getting is to set it as Resource name (`resource "aws_instance" ${NAME}`) and for those this is the regepx they use to validate if it's valid.

Maybe should I rename it for `GetResourceNameFromTag`? Should it be clear?

File: tag/tag_test.go [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+				assert.Len(t, name, 5)
+			} else {
+				assert.Equal(t, tt.Result, name)
+			}
+		})
+	}
+}
+
+// createSRD creates a schema.ResourceData with a
+// 'key' of TypeMap with a 'Name' with 'value'
```
[the remark this answers is not in the document]
Comment (xescugc) [reply tag/tag_test.go]: Yeh I changed the interface after the documentation and I forgot to update it.

Done.

File: writer/hcl.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+	"github.com/cycloidio/terracognita/errcode"
+	"github.com/hashicorp/hcl"
+	"github.com/hashicorp/hcl/hcl/fmtcmd"
+	"github.com/hashicorp/hcl/hcl/printer"
+	"github.com/pkg/errors"
+)
+
+// HCLWriter it's a Writer implementation that writes to
+// a static map to then use to traform to HCL
```
[the remark this answers is not in the document]
Comment (xescugc) [reply writer/hcl.go]: Updated the docs to be more clear

File: writer/hcl.go [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+	if key == "" {
+		return errcode.ErrWriterRequiredKey
+	}
+
+	if value == nil {
+		return errcode.ErrWriterRequiredValue
+	}
+
+	keys := strings.Split(key, ".")
+	if len(keys) != 2 {
```
[the remark this answers is not in the document]
Comment (xescugc) [reply writer/hcl.go]: Correct!. I've added a test+code to prevent it

File: writer/hcl_test.go [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+		)
+
+		err := hw.Write("type.name", value)
+		require.NoError(t, err)
+
+		err = hw.Sync()
+		require.NoError(t, err)
+
+		assert.Equal(t, hcl, b.String())
+	})
```
[the remark this answers is not in the document]
Comment (xescugc) [reply writer/hcl_test.go]: Example of those? At the end the `Sync()` on the HCL one it's just usint the `hcl` lib and the formatter, so there is no much I can test without testing the other libs (the formatter has it's own tests already).

File: writer/state_test.go (hunk 1 above)
Comment (xescugc) [reply writer/state_test.go]: Yes, but for which purpose? what it does behind is just calling Terraform. The only error returned is when encoding the JSON, the rest returns no errors at all. More tests will prove that Terraform works which is not part of this lib.

**Comment by xescugc** [comment]
Rebased from master.

File: tag/tag_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (xlr-8) [reply tag/tag_test.go]: Oh okay, alright makes sense in that case, it's what I'd call the ID of the resource basically, not a tag. But cool, got it.

File: writer/hcl_test.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (xlr-8) [reply writer/hcl_test.go]: Okay, I see

**Pull request merged**